### PR TITLE
feat: allow linting OpenAPI path templates

### PR DESCRIPTION
**Pull request by char0n**
Refs #3517

### CODE REVIEW

File: packages/apidom-ls/src/config/openapi/config.ts [hunk 1]
```diff
@@ -83,4 +84,5 @@ export default {
   tag: tagMeta,
   xml: xmlMeta,
   schema: schemaMeta,
+  'path-template': pathTemplateMeta,
```
Comment (char0n) [inline packages/apidom-ls/src/config/openapi/config.ts:87]: Here we're attaching our meta (containing linting rules) to the `path-template` meta class.

File: packages/apidom-ls/src/config/openapi/path-template/lint/value--valid.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+const valueValidLint: LinterMeta = {
+  code: ApilintCodes.OPENAPI2_PATH_TEMPLATE_VALUE_VALID,
+  source: 'apilint',
+  message: 'path template expressions is not matched with Parameter Object(s)',
+  severity: DiagnosticSeverity.Error,
+  linterFunction: 'apilintOpenAPIPathTemplateValid',
+  marker: 'value',
+  targetSpecs: OpenAPI,
+  conditions: [
```
Comment (char0n) [inline packages/apidom-ls/src/config/openapi/path-template/lint/value--valid.ts:15]: We want to validate matching of Paramater Objects only when the path template is well formed and contains at least one `template expression`. That is achieved by `params: [true]`.

File: packages/apidom-ls/src/services/validation/linter-functions.ts [hunk 3]
```diff
@@ -993,4 +995,23 @@ export const standardLinterfunctions: FunctionItem[] = [
...
+      if (isStringElement(element)) {
+        const pathTemplate = toValue(element);
+        return test(pathTemplate, { strict });
+      }
+      return true;
+    },
+  },
+  {
+    functionName: 'apilintOpenAPIPathTemplateValid',
+    function: (element: Element) => {
```
Comment (char0n) [inline packages/apidom-ls/src/services/validation/linter-functions.ts:1010]: This is where you need to implement the logic of matching template expressions against the Parameter Objects.

File: packages/apidom-ns-openapi-2/src/refractor/visitors/open-api-2/paths/index.ts [hunk 4]
```diff
@@ -26,6 +26,8 @@ const PathsVisitor = stampit(PatternedFieldsVisitor, FallbackVisitor, {
       this.element
         .filter(isPathItemElement)
         .forEach((pathItemElement: PathItemElement, key: StringElement) => {
+          key.classes.push('openapi-path-template');
```
Comment (char0n) [inline packages/apidom-ns-openapi-2/src/refractor/visitors/open-api-2/paths/index.ts:29]: Attaching metadata classes.

File: packages/apidom-ns-openapi-3-0/src/refractor/visitors/open-api-3-0/paths/index.ts [hunk 5]
```diff
@@ -26,6 +26,8 @@ const PathsVisitor = stampit(PatternedFieldsVisitor, FallbackVisitor, {
       this.element
         .filter(isPathItemElement)
         .forEach((pathItemElement: PathItemElement, key: StringElement) => {
+          key.classes.push('openapi-path-template');
```
Comment (char0n) [inline packages/apidom-ns-openapi-3-0/src/refractor/visitors/open-api-3-0/paths/index.ts:29]: Attaching metadata classes.

**Review by kowalczyk-krzysztof — APPROVED** [review]
Great work and thanks for the comments, they helped me a lot 🎉

**Pull request merged**